app/Models/Forum/Topic.php

```diff
@@ -195,6 +195,21 @@ public function nthPost($n)
         return $this->posts()->skip(intval($n) - 1)->first();
     }
 
+    public function isDoublePostBy($userId)
+    {
+        $post = $this->posts()->orderBy('post_id', 'desc')->first();
+        $postTime = $post['post_time'];
+        $now = Carbon::now();
+
+        $diffSinceLastPost = $now->diffInDays($postTime);
+
+        if (! is_null($userId) && $post['poster_id'] == $userId && $diffSinceLastPost < 3) {
+            return true;
```

Comment:
Style choice. Adding a immediate return would have 3 separate returns inside a very short function. This way it is a tick less performant but all validation happens at the same spot. Either it passes -> return true, or it doesn't -> return false. 

Of course I could have it like 
```
    public function isDoublePostBy($userId)
    {
        if (is_null($userId)) {
            return false;
        }

        $post = $this->posts()->orderBy('post_id', 'desc')->first();

        if ($post['poster_id'] != $userId) {
            return false;
        }

        $postTime = $post['post_time'];
        $now = Carbon::now();
        $diffSinceLastPost = $now->diffInDays($postTime);

        if ($diffSinceLastPost < 3) {
            return true;
        }

        return false;
    }
```

but that is a lot longer and harder to read